warehouse/forklift/legacy.py

```diff
@@ -1398,6 +1304,32 @@ def file_upload(request):
                 },
             )
 
+        # If the user provided attestations, verify and store them
+        if "attestations" in request.POST:
+            attestation_service = request.find_service(
+                IReleaseVerificationService, context=None
+            )
+
+            try:
+                attestations: list[Attestation] = (
+                    attestation_service.parse_attestations(
+                        request,
+                        Distribution(name=filename, digest=file_hashes["sha256"]),
+                    )
+                )
+            except AttestationUploadError as e:
+                raise _exc_with_message(
+                    HTTPBadRequest,
+                    str(e),
+                )
+
+            attestation_service.persist_attestations(
+                request.oidc_publisher, attestations, file_
+            )
```

Comment:
```suggestion
        # If the user provided attestations, verify and store them
        if "attestations" in request.POST:
            integrity_service = request.find_service(
                IReleaseVerificationService, context=None
            )

            try:
                attestations: list[Attestation] = (
                    attestation_service.parse_attestations(
                        request,
                        Distribution(name=filename, digest=file_hashes["sha256"]),
                    )
                )
            except AttestationUploadError as e:
                raise _exc_with_message(
                    HTTPBadRequest,
                    str(e),
                )

            integrity_service.persist_attestations(
                request.oidc_publisher, attestations, file_
            )
            provenance = integrity_service.generate_provenance(
                oidc_publisher, attestations
            )
            integrity_service.persist_provenance(provenance, file_)
```
Here's what I'm imagining.